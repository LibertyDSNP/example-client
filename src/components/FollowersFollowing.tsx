import React, { useState } from "react";
import { Button } from "antd";
import FollowersFollowingUsers from "./FollowersFollowingUsers";
import { preFabProfiles } from "../test/testProfiles";

const FollowersFollowing = (): JSX.Element => {
  const [selectedListTitle, setSelectedListTitle] = useState<string>("");

  const tempUserList = [
    {
      profile: preFabProfiles[0],
      icon: preFabProfiles[0].icon,
      name: preFabProfiles[0].name,
      following: true,
      followsMe: true,
    },
    {
      profile: preFabProfiles[1],
      icon: preFabProfiles[1].icon,
      name: preFabProfiles[1].name,
      following: true,
      followsMe: false,
    },
    {
      profile: preFabProfiles[2],
      icon: preFabProfiles[2].icon,
      name: preFabProfiles[2].name,
      following: false,
      followsMe: true,
    },
  ];

  const handleClick = (listTitle: string) => {
    if (selectedListTitle === listTitle) {
      setSelectedListTitle("");
      return;
    }
    if (listTitle === "showFollowers") {
      setSelectedListTitle(listTitle);
    } else {
      setSelectedListTitle(listTitle);
    }
  };

  const getClassName = (name: string) => {
    return selectedListTitle === name
      ? "FollowersFollowing__button FollowersFollowing__button--active"
      : "FollowersFollowing__button";
  };

  const followerCount = tempUserList.filter((user) => user.followsMe).length;
  const followingCount = tempUserList.filter((user) => user.following).length;

  return (
    <div className="FollowersFollowing__block">
      <div className="FollowersFollowing__buttonBlock">
        <Button
          className={getClassName("showFollowers")}
          onClick={() => handleClick("showFollowers")}
        >
          <div className="FollowersFollowing__buttonCount">{followerCount}</div>
          Followers
        </Button>
        {selectedListTitle !== "showFollowing" &&
          selectedListTitle !== "showFollowers" && (
            <div className="FollowersFollowing__buttonSeperator"> </div>
          )}
        <Button
          className={getClassName("showFollowing")}
          onClick={() => handleClick("showFollowing")}
        >
          <div className="FollowersFollowing__buttonCount">
            {followingCount}
          </div>
          Following
        </Button>
      </div>
      <FollowersFollowingUsers
        selectedListTitle={selectedListTitle}
        tempUserList={tempUserList}
      />
    </div>
  );
};

export default FollowersFollowing;
