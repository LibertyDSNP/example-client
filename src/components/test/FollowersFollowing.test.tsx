import React from "react";
import FollowersFollowing from "../FollowersFollowing";
import { shallow } from "enzyme";

describe("Feed", () => {
  it("renders without crashing", () => {
    expect(() => {
      shallow(<FollowersFollowing />);
    }).not.toThrow();
  });
  it("to match snapshot", () => {
    //write more specific tests when we are pulling from a real followers/following list
    //I want to test more specific actions
    const component = shallow(<FollowersFollowing />);
    expect(component).toMatchSnapshot();
  });
});
