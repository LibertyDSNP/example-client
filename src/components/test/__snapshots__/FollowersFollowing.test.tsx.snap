// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed to match snapshot 1`] = `
<div
  className="FollowersFollowing__block"
>
  <div
    className="FollowersFollowing__buttonBlock"
  >
    <Button
      className="FollowersFollowing__button"
      onClick={[Function]}
    >
      <div
        className="FollowersFollowing__buttonCount"
      >
        2
      </div>
      Followers
    </Button>
    <div
      className="FollowersFollowing__buttonSeperator"
    >
       
    </div>
    <Button
      className="FollowersFollowing__button"
      onClick={[Function]}
    >
      <div
        className="FollowersFollowing__buttonCount"
      >
        2
      </div>
      Following
    </Button>
  </div>
  <FollowersFollowingUsers
    selectedListTitle=""
    tempUserList={
      Array [
        Object {
          "following": true,
          "followsMe": true,
          "icon": Object {
            "url": "https://image.shutterstock.com/image-vector/monday-time-sparkle-shine-word-600w-731826949.jpg",
          },
          "name": "Monday January",
          "profile": Object {
            "@context": "https://www.w3.org/ns/activitystreams",
            "actor": "0x1DCODE0000",
            "discoverable": true,
            "icon": Object {
              "url": "https://image.shutterstock.com/image-vector/monday-time-sparkle-shine-word-600w-731826949.jpg",
            },
            "id": "http://localhost:3003/api/announce/0x1DCODE0000",
            "name": "Monday January",
            "preferredUsername": "0Monday0",
            "socialAddress": "0x1DCODE0000",
            "summary": "",
            "type": "Person",
            "url": "",
            "walletAddress": "0xCODE0000",
          },
        },
        Object {
          "following": true,
          "followsMe": false,
          "icon": Object {
            "url": "https://image.shutterstock.com/image-vector/traditional-taco-tuesday-neon-light-600w-1193206603.jpg",
          },
          "name": "Tuesday February",
          "profile": Object {
            "@context": "https://www.w3.org/ns/activitystreams",
            "actor": "0x1DCODE1111",
            "discoverable": true,
            "icon": Object {
              "url": "https://image.shutterstock.com/image-vector/traditional-taco-tuesday-neon-light-600w-1193206603.jpg",
            },
            "id": "http://localhost:3003/api/announce/0x1DCODE1111",
            "name": "Tuesday February",
            "preferredUsername": "1Tuesday1",
            "socialAddress": "0x1DCODE1111",
            "summary": "",
            "type": "Person",
            "url": "",
            "walletAddress": "0xCODE1111",
          },
        },
        Object {
          "following": false,
          "followsMe": true,
          "icon": Object {
            "url": "https://image.shutterstock.com/image-vector/wednesday-pop-art-illustration-vector-600w-219333010.jpg",
          },
          "name": "Wednesday March",
          "profile": Object {
            "@context": "https://www.w3.org/ns/activitystreams",
            "actor": "0x1DCODE2222",
            "discoverable": true,
            "icon": Object {
              "url": "https://image.shutterstock.com/image-vector/wednesday-pop-art-illustration-vector-600w-219333010.jpg",
            },
            "id": "http://localhost:3003/api/announce/0x1DCODE2222",
            "name": "Wednesday March",
            "preferredUsername": "2Wednesday2",
            "socialAddress": "0x1DCODE2222",
            "summary": "",
            "type": "Person",
            "url": "",
            "walletAddress": "0xCODE2222",
          },
        },
      ]
    }
  />
</div>
`;
